Sctp fixes

fabbione opened the pull request.

chrissie-c commented:
Would it be reasonable to only enable the use of send() on platforms where it's absolutely necessary? I'm a little unhappy with the copying in _sendmmsg() happening on Linux

fabbione commented:
@chrissie-c yes I had the same idea while I was driving back home and I was hoping to catch you before you left for the weekend :P

fabbione commented:
retest this please

fabbione committed:
[logging] fix log target of recently added API calls

spotted during sctp testing

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione commented:
retest this please

fabbione committed:
[transport] don´t specify destination address for connection oriented protocols

some implementations of sendto/sendmsg expects dst_addr / msg_name to be
empty for connection oriented protocols such as SCTP (spotted on FreeBSD).

Make sure to clear or not pass those information on connection oriented protocol.

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione commented:
@chrissie-c brand new patch, much better way to fix the issue and much less invasive.

fabbione commented:
retest this please

chrissie-c reviewed (approved):
That's much nicer.
ACK
